meta-ros2-humble/recipes-core/packagegroups/packagegroup-ros-world-humble.bb

```diff
@@ -198,25 +227,61 @@ ROS_SUPERFLORE_GENERATED_WORLD_PACKAGES_DEPENDING_ON_OPENGL = " \
 
 RDEPENDS:${PN}:remove = "${@bb.utils.contains('ROS_WORLD_SKIP_GROUPS', 'qt5', '${ROS_SUPERFLORE_GENERATED_WORLD_PACKAGES_DEPENDING_ON_META_QT5}', '', d)}"
 ROS_SUPERFLORE_GENERATED_WORLD_PACKAGES_DEPENDING_ON_META_QT5 = " \
+    cartographer-rviz \
```

Comment:
I'm wondering whether it is possible to figure out how many ROS packages from the [`distribution.yaml`](https://github.com/ros/rosdistro/blob/humble/2022-11-23/humble/distribution.yaml) got built with `ros-image-world`?

Reply:
There are a couple of ways.  You can use the -g option to bitbake to create a graphviz dot file of the task dependencies (task-depends.dot) or a text file with the package names (pn-buildlist). (eg. bitbake -g ros-image-world )

There is also a manifest file in tmp-glibc/deploy/images/raspberrypi4-64/ros-image-world-humble-raspberrypi4-64.manifest that will tell you the packages that were produced.

I think you want pn-buildlist to do your comparison to distribution.yaml.

Reply:
thanks @robwoolley. Is the number of generated `bb` files corresponds 1:1 to the `distribution.yaml`?

```bash
akholodn@razilix:~/yws/meta-ros$ find meta-ros2-humble/generated-recipes/ -name *.bb | wc
   1017    1017   78196
```

Reply:
You would need to filter out those packages that are part of oe-core itself.  There are only around 415 packages listed in distribution.yaml.

Reply:
I count 1017 `bb` files
```bash
$ find meta-ros/meta-ros2-humble/generated-recipes/ -name *.bb | wc
   1017    1017   17735
```

```bash
$ bitbake -g ros-image-world
$ cat pn-buildlist | sort > world.txt
$ find meta-ros/meta-ros2-humble/generated-recipes/ -name *.bb | rev | cut -d '_' -f 2 | cut -d '/' -f 1 | rev | sort > generated.txt
$ awk 'FNR==NR {a[$0]++; next} !($0 in a)' world.txt generated.txt > missing.txt
$ cat missing.txt | wc
    253     253    4286
```